pkg/secretless/config/v2/config.go

```diff
@@ -0,0 +1,120 @@
+// v2 is package for parsing version 2 "secretless.yml" files and converting
+// them into "v1.Config" objects.  Users of this package will typically only be
+// concerned with calling "NewV1Config" to parse
+package v2
+
+import (
+	"github.com/cyberark/secretless-broker/pkg/secretless/config/v1"
+	"gopkg.in/yaml.v2"
+	"sort"
+)
+
+// Config represents a full configuration of Secretless, which is just a list of
+// Service configurations.
+type Config struct {
+	Services []*Service
+}
+
+// Service represents a the configuration of a Secretless proxy service. It
+// includes the services protocol, where it's listening, where to find its
+// required credentials, and (optionally) any additional protocol specific
+// configuration.
+type Service struct {
+	Name           string
+	Credentials    []*Credential
+	Protocol       string
+	ListenOn       string
+	ProtocolConfig []byte
+}
+
+// NewV1Config is converts the bytes of a v2 YAML file to a v1.Config.  As such,
+// it's the primary public interface of the v2 package.
+// TODO: Possible move to v1.config?
+func NewV1Config(v2YAML []byte) (*v1.Config, error) {
+	v2cfg, err := NewConfig(v2YAML)
```

Comment:
Not a blocker but it's a bit of a dense var name where `v2Config` might be a bit better for readability